plugins/org.eclipse.fordiac.ide.attributetypeeditor/src/org/eclipse/fordiac/ide/attributetypeeditor/properties/AttributeTargetSection.java

```diff
@@ -83,32 +84,19 @@ public void createCheckBoxes(final Composite parent) {
 		composite.setLayout(new GridLayout(7, false));
 		composite.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
 
-		final Composite ungroupedComposite = getWidgetFactory().createComposite(composite);
-		ungroupedComposite.setData(TARGET_CATEGORY, AttributeTarget.EMPTY_GROUP);
-		ungroupedComposite.setLayout(new GridLayout(1, false));
-		ungroupedComposite.setLayoutData(new GridData(SWT.FILL, SWT.TOP, true, false));
-
-		final Map<String, Composite> groups = new HashMap<>();
+		final Map<String, Group> groups = new HashMap<>();
 		final StructuredType targetType = (StructuredType) InternalAttributeDeclarations.TARGET.getType();
 
 		targetType.getMemberVariables().stream().map(member -> AttributeTarget.fromName(member.getName()))
 				.filter(Objects::nonNull).forEach(target -> {
-					final String category = target.getCategory();
-					Composite container = composite;
-
-					if (category.equals(AttributeTarget.EMPTY_GROUP)) {
-						container = ungroupedComposite;
-					} else {
-						container = groups.computeIfAbsent(category, cat -> {
-							final Composite group = getWidgetFactory().createComposite(composite);
-							group.setLayout(new GridLayout(1, false));
-							group.setLayoutData(new GridData(SWT.FILL, SWT.TOP, true, false));
-
-							getWidgetFactory().createLabel(group, cat);
-							group.setData(TARGET_CATEGORY, cat);
-							return group;
-						});
-					}
+					final Group container = groups.computeIfAbsent(target.getCategory(), cat -> {
+						final Group group = getWidgetFactory().createGroup(composite, cat);
+						group.setLayout(new GridLayout(1, false));
+						group.setLayoutData(new GridData(SWT.FILL, SWT.TOP, true, false));
```

Comment:
If you use SWT.FILL instead of TOP the groups would have the same height which looks more harmonic.